### ISSUE TITLE
Training slow in Azure DSVM (Standard NC6)

### ISSUE THREAD

**Issue by yangmingwanli** [microsoft/nni#3568]
**Environment**:
- NNI version:  2.1
- NNI mode (local|remote|pai):  local
- Client OS:  Ubuntu 18.04
- Server OS (for remote mode only):
- Python version:  Python 3.6.9
- PyTorch/TensorFlow version: PyTorch 1.4.0
- Is conda/virtualenv/venv used?: conda (env name azureml_py36_pytorch)
- Is running in Docker?:  No

**Log message**:
 - nnimanager.log: 
[2021-04-22 06:53:44] INFO [ 'Datastore initialization done' ]
[2021-04-22 06:53:44] INFO [ 'RestServer start' ]
[2021-04-22 06:53:44] INFO [ 'Construct local machine training service.' ]
[2021-04-22 06:53:44] INFO [ 'RestServer base port is 8080' ]
[2021-04-22 06:53:44] INFO [ 'Rest server listening on: http://0.0.0.0:8080' ]
[2021-04-22 06:53:45] INFO [ 'NNIManager setClusterMetadata, key: local_config, value: {"useActiveGpu":true,"gpuIndices":"0","maxTrialNumPerGpu":1}' ]
[2021-04-22 06:53:45] INFO [ 'Specified GPU indices: 0' ]
[2021-04-22 06:53:45] INFO [ 'NNIManager setClusterMetadata, key: trial_config, value: {"command":"python3 mnist.py","codeDir":"/home/yangming/nni/examples/trials/mnist-pytorch/.","gpuNum":1}' ]
[2021-04-22 06:53:45] INFO [ 'required GPU number is 1' ]
[2021-04-22 06:53:45] INFO [ 'Starting experiment: ZezDh0pJ' ]
[2021-04-22 06:53:45] INFO [ 'Change NNIManager status from: INITIALIZED to: RUNNING' ]
[2021-04-22 06:53:45] INFO [ 'Add event listeners' ]
[2021-04-22 06:53:45] INFO [ 'Run local machine training service.' ]
[2021-04-22 06:53:45] WARNING [ 'gpu_metrics file does not exist!' ]
[2021-04-22 06:53:46] INFO [ 'NNIManager received command from dispatcher: ID, ' ]
[2021-04-22 06:53:46] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 0, "parameter_source": "algorithm", "parameters": {"batch_size": 64, "hidden_size": 512, "lr": 0.01, "momentum": 0.26083012832194785}, "parameter_index": 0}' ]
[2021-04-22 06:53:50] INFO [ 'submitTrialJob: form: {"sequenceId":0,"hyperParameters":{"value":"{\\"parameter_id\\": 0, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"batch_size\\": 64, \\"hidden_size\\": 512, \\"lr\\": 0.01, \\"momentum\\": 0.26083012832194785}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-04-22 06:53:50] WARNING [ 'gpu_metrics file does not exist!' ]
[2021-04-22 06:54:00] INFO [ 'Trial job eB4Nj status changed from WAITING to RUNNING' ]
[2021-04-22 06:56:41] INFO [ 'Trial job eB4Nj status changed from RUNNING to SUCCEEDED' ]
[2021-04-22 06:56:41] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 1, "parameter_source": "algorithm", "parameters": {"batch_size": 128, "hidden_size": 512, "lr": 0.1, "momentum": 0.97466872750186}, "parameter_index": 0}' ]
[2021-04-22 06:56:46] INFO [ 'submitTrialJob: form: {"sequenceId":1,"hyperParameters":{"value":"{\\"parameter_id\\": 1, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"batch_size\\": 128, \\"hidden_size\\": 512, \\"lr\\": 0.1, \\"momentum\\": 0.97466872750186}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-04-22 06:56:51] INFO [ 'Trial job xUbJF status changed from WAITING to RUNNING' ]
[2021-04-22 06:59:06] INFO [ 'Trial job xUbJF status changed from RUNNING to SUCCEEDED' ]
[2021-04-22 06:59:06] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 2, "parameter_source": "algorithm", "parameters": {"batch_size": 32, "hidden_size": 1024, "lr": 0.1, "momentum": 0.2227634080159795}, "parameter_index": 0}' ]
[2021-04-22 06:59:11] INFO [ 'submitTrialJob: form: {"sequenceId":2,"hyperParameters":{"value":"{\\"parameter_id\\": 2, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"batch_size\\": 32, \\"hidden_size\\": 1024, \\"lr\\": 0.1, \\"momentum\\": 0.2227634080159795}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-04-22 06:59:16] INFO [ 'Trial job lgSSE status changed from WAITING to RUNNING' ]
[2021-04-22 07:01:51] INFO [ 'Trial job lgSSE status changed from RUNNING to SUCCEEDED' ]
[2021-04-22 07:01:51] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 3, "parameter_source": "algorithm", "parameters": {"batch_size": 16, "hidden_size": 256, "lr": 0.001, "momentum": 0.7389797150765158}, "parameter_index": 0}' ]
[2021-04-22 07:01:56] INFO [ 'submitTrialJob: form: {"sequenceId":3,"hyperParameters":{"value":"{\\"parameter_id\\": 3, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"batch_size\\": 16, \\"hidden_size\\": 256, \\"lr\\": 0.001, \\"momentum\\": 0.7389797150765158}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-04-22 07:02:01] INFO [ 'Trial job IyDe6 status changed from WAITING to RUNNING' ]
[2021-04-22 07:05:17] INFO [ 'Trial job IyDe6 status changed from RUNNING to SUCCEEDED' ]
[2021-04-22 07:05:17] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 4, "parameter_source": "algorithm", "parameters": {"batch_size": 64, "hidden_size": 256, "lr": 0.1, "momentum": 0.34030864605620104}, "parameter_index": 0}' ]
[2021-04-22 07:05:22] INFO [ 'submitTrialJob: form: {"sequenceId":4,"hyperParameters":{"value":"{\\"parameter_id\\": 4, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"batch_size\\": 64, \\"hidden_size\\": 256, \\"lr\\": 0.1, \\"momentum\\": 0.34030864605620104}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-04-22 07:05:27] INFO [ 'Trial job pTEjG status changed from WAITING to RUNNING' ]
[2021-04-22 07:07:47] INFO [ 'Trial job pTEjG status changed from RUNNING to SUCCEEDED' ]
[2021-04-22 07:07:47] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 5, "parameter_source": "algorithm", "parameters": {"batch_size": 16, "hidden_size": 1024, "lr": 0.001, "momentum": 0.6297666470259692}, "parameter_index": 0}' ]
[2021-04-22 07:07:52] INFO [ 'submitTrialJob: form: {"sequenceId":5,"hyperParameters":{"value":"{\\"parameter_id\\": 5, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"batch_size\\": 16, \\"hidden_size\\": 1024, \\"lr\\": 0.001, \\"momentum\\": 0.6297666470259692}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-04-22 07:07:57] INFO [ 'Trial job zURL6 status changed from WAITING to RUNNING' ]
[2021-04-22 07:11:03] INFO [ 'Trial job zURL6 status changed from RUNNING to SUCCEEDED' ]
[2021-04-22 07:11:03] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 6, "parameter_source": "algorithm", "parameters": {"batch_size": 128, "hidden_size": 128, "lr": 0.0001, "momentum": 0.9875695155975203}, "parameter_index": 0}' ]
[2021-04-22 07:11:08] INFO [ 'submitTrialJob: form: {"sequenceId":6,"hyperParameters":{"value":"{\\"parameter_id\\": 6, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"batch_size\\": 128, \\"hidden_size\\": 128, \\"lr\\": 0.0001, \\"momentum\\": 0.9875695155975203}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-04-22 07:11:13] INFO [ 'Trial job w6Nf3 status changed from WAITING to RUNNING' ]
[2021-04-22 07:13:28] INFO [ 'Trial job w6Nf3 status changed from RUNNING to SUCCEEDED' ]
[2021-04-22 07:13:28] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 7, "parameter_source": "algorithm", "parameters": {"batch_size": 128, "hidden_size": 256, "lr": 0.1, "momentum": 0.5646997072218127}, "parameter_index": 0}' ]
[2021-04-22 07:13:33] INFO [ 'submitTrialJob: form: {"sequenceId":7,"hyperParameters":{"value":"{\\"parameter_id\\": 7, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"batch_size\\": 128, \\"hidden_size\\": 256, \\"lr\\": 0.1, \\"momentum\\": 0.5646997072218127}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-04-22 07:13:38] INFO [ 'Trial job Taxg3 status changed from WAITING to RUNNING' ]
[2021-04-22 07:15:48] INFO [ 'Trial job Taxg3 status changed from RUNNING to SUCCEEDED' ]
[2021-04-22 07:15:48] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 8, "parameter_source": "algorithm", "parameters": {"batch_size": 32, "hidden_size": 128, "lr": 0.01, "momentum": 0.5836579252570034}, "parameter_index": 0}' ]
[2021-04-22 07:15:53] INFO [ 'submitTrialJob: form: {"sequenceId":8,"hyperParameters":{"value":"{\\"parameter_id\\": 8, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"batch_size\\": 32, \\"hidden_size\\": 128, \\"lr\\": 0.01, \\"momentum\\": 0.5836579252570034}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-04-22 07:15:58] INFO [ 'Trial job Dasbc status changed from WAITING to RUNNING' ]
[2021-04-22 07:18:35] INFO [ 'Trial job Dasbc status changed from RUNNING to SUCCEEDED' ]
[2021-04-22 07:18:35] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 9, "parameter_source": "algorithm", "parameters": {"batch_size": 128, "hidden_size": 256, "lr": 0.01, "momentum": 0.22826712320236786}, "parameter_index": 0}' ]
[2021-04-22 07:18:40] INFO [ 'submitTrialJob: form: {"sequenceId":9,"hyperParameters":{"value":"{\\"parameter_id\\": 9, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"batch_size\\": 128, \\"hidden_size\\": 256, \\"lr\\": 0.01, \\"momentum\\": 0.22826712320236786}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-04-22 07:18:45] INFO [ 'Trial job mMkzh status changed from WAITING to RUNNING' ]
[2021-04-22 07:18:45] INFO [ 'Change NNIManager status from: RUNNING to: NO_MORE_TRIAL' ]
[2021-04-22 07:20:55] INFO [ 'Trial job mMkzh status changed from RUNNING to SUCCEEDED' ]
[2021-04-22 07:20:55] INFO [ 'Change NNIManager status from: NO_MORE_TRIAL to: DONE' ]
[2021-04-22 07:20:55] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 10, "parameter_source": "algorithm", "parameters": {"batch_size": 64, "hidden_size": 128, "lr": 0.01, "momentum": 0.5179615389153926}, "parameter_index": 0}' ]
[2021-04-22 07:20:55] INFO [ 'Experiment done.' ]

 - dispatcher.log:
 [2021-04-22 06:53:46] INFO (nni.runtime.msg_dispatcher_base/MainThread) Dispatcher started
[2021-04-22 06:53:46] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.001722 seconds
[2021-04-22 06:53:46] INFO (hyperopt.tpe/Thread-1) TPE using 0 trials
[2021-04-22 06:56:41] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.001892 seconds
[2021-04-22 06:56:41] INFO (hyperopt.tpe/Thread-1) TPE using 1/1 trials with best loss -98.810000
[2021-04-22 06:59:06] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.001930 seconds
[2021-04-22 06:59:06] INFO (hyperopt.tpe/Thread-1) TPE using 2/2 trials with best loss -98.810000
[2021-04-22 07:01:51] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.001872 seconds
[2021-04-22 07:01:51] INFO (hyperopt.tpe/Thread-1) TPE using 3/3 trials with best loss -99.200000
[2021-04-22 07:05:17] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.001886 seconds
[2021-04-22 07:05:17] INFO (hyperopt.tpe/Thread-1) TPE using 4/4 trials with best loss -99.200000
[2021-04-22 07:07:47] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.001870 seconds
[2021-04-22 07:07:47] INFO (hyperopt.tpe/Thread-1) TPE using 5/5 trials with best loss -99.200000
[2021-04-22 07:11:03] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.002196 seconds
[2021-04-22 07:11:03] INFO (hyperopt.tpe/Thread-1) TPE using 6/6 trials with best loss -99.200000
[2021-04-22 07:13:28] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.001897 seconds
[2021-04-22 07:13:28] INFO (hyperopt.tpe/Thread-1) TPE using 7/7 trials with best loss -99.200000
[2021-04-22 07:15:48] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.002009 seconds
[2021-04-22 07:15:48] INFO (hyperopt.tpe/Thread-1) TPE using 8/8 trials with best loss -99.200000
[2021-04-22 07:18:35] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.001865 seconds
[2021-04-22 07:18:35] INFO (hyperopt.tpe/Thread-1) TPE using 9/9 trials with best loss -99.200000
[2021-04-22 07:20:55] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.001810 seconds
[2021-04-22 07:20:55] INFO (hyperopt.tpe/Thread-1) TPE using 10/10 trials with best loss -99.200000

 - nnictl stdout and stderr:
 -----------------------------------------------------------------------
                Experiment start time 2021-04-22 06:53:43
-----------------------------------------------------------------------

-----------------------------------------------------------------------
                Experiment start time 2021-04-22 06:53:43
-----------------------------------------------------------------------
/anaconda/envs/azureml_py36_pytorch/lib/python3.6/site-packages/json_tricks/nonp.py:225: JsonTricksDeprecation: `json_tricks.load(s)` stripped some comments, but `ignore_comments` was not passed; in the next major release, the behaviour when `ignore_comments` is not passed will change; it is recommended to explicitly pass `ignore_comments=True` if you want to strip comments; see https://github.com/mverleg/pyjson_tricks/issues/74
  JsonTricksDeprecation)
Error: Dispatcher stream error, tuner may have crashed.
    at EventEmitter.dispatcher.onError (/anaconda/envs/azureml_py36_pytorch/lib/python3.6/site-packages/nni_node/core/nnimanager.js:550:32)
    at EventEmitter.emit (events.js:198:13)
    at Socket.IpcInterface.outgoingStream.on (/anaconda/envs/azureml_py36_pytorch/lib/python3.6/site-packages/nni_node/core/ipcInterface.js:42:72)
    at Socket.emit (events.js:198:13)
    at errorOrDestroy (internal/streams/destroy.js:107:12)
    at onwriteError (_stream_writable.js:430:5)
    at onwrite (_stream_writable.js:461:5)
    at _destroy (internal/streams/destroy.js:49:7)
    at Socket._destroy (net.js:614:3)
    at Socket.destroy (internal/streams/destroy.js:37:8)

 

**What issue meet, what's expected?**:
Run mnist-pytorch example on a GPU sized Azure DSVM, training took much logger than expected. 

**How to reproduce it?**: 
Change nni/examples/trials/mnist-pytorch/config.yml as shown below and run the expriment
authorName: default
experimentName: example_mnist_pytorch
trialConcurrency: 1
maxExecDuration: 1h
maxTrialNum: 10
#choice: local, remote, pai
trainingServicePlatform: local
searchSpacePath: search_space.json
#choice: true, false
useAnnotation: false
tuner:
  #choice: TPE, Random, Anneal, Evolution, BatchTuner, MetisTuner, GPTuner
  #SMAC (SMAC should be installed through nnictl)
  builtinTunerName: TPE
  classArgs:
    #choice: maximize, minimize
    optimize_mode: maximize
trial:
  command: python3 mnist.py
  codeDir: .
  gpuNum: 1
localConfig:
  useActiveGpu: true
  gpuIndices: "0"
  maxTrialNumPerGpu: 1

**Additional information**:
/tmp/username/nni folder is empty, no gpu_metrics file
nvidia-smi shows GPU memory usage at 345MiB / 11441MiB, GPU-Util at 51%, python3 is listed under Processes.

**Comment by SparkSnail** [comment]
Add @acured to take a look.

**Comment by acured** [comment]
Hi @yangmingwanli,
Could you double check whether GPU is used? You can check it from whether there is a difference on "nvidia-smi".

Under the guarantee of GPU usage, it will take almost 90s for each trail, almost the same with yours. Acutely the example's network and search space is very small, the speed advantage of GPU is not obvious, you can try to modify those search space  parameter to see the effect.

**Comment by yangmingwanli** [comment]
Thank you @acured for the explanations.
